### DISCUSSION TITLE
Toggle completions on demand

### DISCUSSION THREAD

**Discussion by nounder** [saghen/blink.cmp#608] [Q&A]
### Feature Description

I'd like to toggle the plugin globally or per buffer.

It's seems like a very basic functionality. I've searched for it using `:h blink` but couldn't find anything.

**Comment by saghen** [comment]
Sounds like you're looking for `enabled = function() ... end`. More info in the `default configuration` section of the readme. We don't have vim docs yet

**Comment by dearieme** [comment]
I have same question, I have a keymap for nvim-cmp to toggle completions for current buffer, and would like similar with blink but can't figure out how. Looking at the enabled function in the default config (and without really knowing what I'm doing)I tried:
```
vim.keymap.set("n", "<leader>bc", function()
  vim.api.nvim_buf_set_var(0, "completion", false)
end, { desc = "Disable blink.cmp for current buffer" })
```
The buffer var gets updated, but completions are still enabled. Any help appreciated, cheers.

**Reply by saghen** [reply]
I've opened #622 which I believe is the issue you're running into

**Comment by dearieme** [comment]
Snippet above now works for me, thanks!

**Reply by rieje** [reply]
How to toggle? I tried getting the state with `vim.api.nvim_buf_get_var(0, "completion")` as in:

    vim.keymap.set("n", "<leader>bc", function()
      if vim.api.nvim_buf_get_var(0, "completion") then
        vim.api.nvim_buf_set_var(0, "completion", false)
      else
        vim.api.nvim_buf_set_var(0, "completion", true)
      end
    end, { desc = "Toggle blink.cmp completions current buffer" })

but get `Key not found: completion`. 

Also curious to know if it's possible to only toggle a particular source, e.g. [dictionary](https://github.com/Kaiser-Yang/blink-cmp-dictionary) or restrict it to using only on a particular file.

**Reply by khaosdoctor** [reply]
So I had this same problem and I didn't find any specific solution to this. So I basically overrode the function and created my own variable:

```lua
function()
  local blink = require("blink.cmp.config")
  local blinkVarName = "blink_enabled"

  blink.enabled = function(checkOnly)
    -- Prevent running the function when checking
    checkOnly = checkOnly == nil and true or checkOnly
   
    -- Error handling
    local currentStatus
    if not checkOnly then
      if pcall(vim.api.nvim_buf_get_var, 0, blinkVarName) then
        currentStatus = vim.api.nvim_buf_get_var(0, blinkVarName)
      else
        currentStatus = true
      end

      vim.api.nvim_buf_set_var(0, blinkVarName, not currentStatus)

      local valueString = vim.api.nvim_buf_get_var(0, blinkVarName) and "enabled" or "disabled"
      require("snacks.notify")(
        "Blink completions are now " .. valueString,
        { level = "info", title = "Completions toggled" }
      )
    end

    return vim.api.nvim_buf_get_var(0, blinkVarName)
  end
  blink.enabled(false)
end
```

You can set this either in the keymap, but you can also configure blink to start with this function as the `enabled` configuration in the setup in your plugin file
